Realm/RLMLogger.mm

```diff
@@ -120,10 +172,26 @@ - (instancetype)initWithLogger:(std::shared_ptr<Logger>)logger {
     return self;
 }
 
-- (instancetype)initWithLevel:(RLMLogLevel)level logFunction:(RLMLogFunction)logFunction {
+- (instancetype)initWithLevel:(RLMLogLevel)level
+                  logFunction:(RLMLogFunction)logFunction {
     if (self = [super init]) {
         auto logger = std::make_shared<CustomLogger>();
         logger->set_level_threshold(levelForLogLevel(level));
+        auto block = [logFunction](RLMLogLevel level, RLMLogCategory, NSString *message) {
+            logFunction(level, message);
+        };
+        logger->function = block;
+        self->_logger = logger;
+    }
+    return self;
+}
+
+- (instancetype)initWithLevel:(RLMLogLevel)level
+                     category:(RLMLogCategory)category
```

Comment:
The loggers created with this function are of the default category `realm`. If you like to filter by category you might need to instantiate a Logger with the category.

Reply:
Changed the approach, and now you can only set the global log level for a category using setLogLevel. initWithLevel is gonna is deprecated